src/psyclone/dynamo0p3.py

```diff
@@ -2669,26 +2669,22 @@ class DynFields(DynCollection):
     def _invoke_declarations(self, parent):
         '''
         Add field-related declarations to the PSy-layer routine.
+        Note: PSy layer in LFRic does not modify the field objects. Hence,
+        their Fortran intents are always in (the data pointed to by the
+        fields are updated in kernels).
 
         :param parent: the node in the f2pygen AST representing the PSy-layer \
                        routine to which to add declarations.
         :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
         '''
         from psyclone.f2pygen import TypeDeclGen
-        # Add the subroutine argument declarations for fields
-        fld_args = self._invoke.unique_declns_by_intent("gh_field")
-        for intent in FORTRAN_INTENT_NAMES:
-            if fld_args[intent]:
-                if intent == "out":
-                    # The data part of a field might have intent(out) but
-                    # in order to preserve the state of the whole
-                    # derived-type object it must be declared as inout.
-                    fort_intent = "inout"
-                else:
-                    fort_intent = intent
-                parent.add(TypeDeclGen(parent, datatype="field_type",
-                                       entity_decls=fld_args[intent],
-                                       intent=fort_intent))
+
+        # Add the Invoke subroutine argument declarations for fields
```

Comment:
Please extend this comment to explain why we can safely declare all field objects to have intent in.

Reply:
For consistency with similar comments I left this one as as and updated the `_invoke_declarations` docstring along the lines of the one for operators.